File: backend/app/schemas/material.py
"""Material 도메인 DTO"""
from __future__ import annotations

from datetime import datetime
from decimal import Decimal
from typing import Optional
from uuid import UUID

from pydantic import BaseModel


class MaterialResponse(BaseModel):
    id: UUID
    material_code: str
    material_name: str
    category: Optional[str] = None
    is_active: bool
    created_at: datetime


class MaterialRfProfileResponse(BaseModel):
    id: UUID
    material_id: UUID
    freq_ghz: Decimal
    permittivity: Decimal
    conductivity: Decimal
    penetration_loss_db: Decimal
    reference_thickness_m: Decimal
    profile_version: int
    is_default: bool

File: frontend/src/types/material.ts
import type { ISODateString, UUID } from './common';

// 백엔드 §12 Material DTO 매핑 (app/schemas/material.py)

export interface Material {
  id: UUID;
  material_code: string;
  material_name: string;
  category: string | null;
  is_active: boolean;
  created_at: ISODateString;
}

export interface MaterialRfProfile {
  id: UUID;
  material_id: UUID;
  freq_ghz: string;
  permittivity: string;
  conductivity: string;
  penetration_loss_db: string;
  reference_thickness_m: string;
  profile_version: number;
  is_default: boolean;
}
